src/main/java/marquez/service/JobService.java

```diff
@@ -242,13 +260,52 @@ public Job createOrUpdate(
     }
   }
 
+  private void updateRunInputDatasets(
+      UUID runUuid,
+      List<DatasetRow> inputRows,
+      NamespaceName namespaceName,
+      JobName jobName,
+      JobVersionRow jobVersion) {
+    if (inputRows.isEmpty()) {
+      return;
+    }
+    List<RunInput> runInputs = new ArrayList<>();
+    for (DatasetRow inputDataset : inputRows) {
+      Optional<DatasetVersionRow> version =
+          datasetVersionDao.findLatestDatasetByUuid(inputDataset.getUuid());
+      if (version.isPresent()) {
+        DatasetVersionRow row = version.get();
+        runDao.updateInputVersions(runUuid, version.get().getUuid());
+        runInputs.add(
+            new RunInput(
+                new DatasetVersionId(
+                    namespaceName, DatasetName.of(inputDataset.getName()), row.getUuid())));
+      }
+    }
+    Optional<ExtendedRunRow> runRow = runDao.findBy(runUuid);
+    if (runRow.isEmpty()) {
```

Comment:
should we log an error in that case?

Reply:
actually on second thoughts we might want to throw an error if the run is not found here (same for comment bellow).
The api does that validation too so it wouldn't be a reachable case. It is still better to throw an exception so that if things change in the future it is obvious what's happening:
https://github.com/MarquezProject/marquez/blob/c927c88d17edd1b65f57d5428d441f6c0886ec13/src/main/java/marquez/api/JobResource.java#L87